### PR TITLE
Adds linter to ci

### PR DESCRIPTION
**Pull request by sirkrypt0**
This PR adds the custom made linter to the CI and appeases it.
Depends on #36

### CODE REVIEW

File: kernel/network/e1000.h [hunk 1]
```diff
@@ -71,31 +71,31 @@
...
-#define RCTL_MO_32                      (3 << 12)   // Multicast Offset - bits 43:32
-#define RCTL_BAM                        (1 << 15)   // Broadcast Accept Mode
-#define RCTL_VFE                        (1 << 18)   // VLAN Filter Enable
-#define RCTL_CFIEN                      (1 << 19)   // Canonical Form Indicator Enable
-#define RCTL_CFI                        (1 << 20)   // Canonical Form Indicator Bit Value
-#define RCTL_DPF                        (1 << 22)   // Discard Pause Frames
-#define RCTL_PMCF                       (1 << 23)   // Pass MAC Control Frames
-#define RCTL_SECRC                      (1 << 26)   // Strip Ethernet CRC
+#define REG_TIPG         0x0410 // Transmit Inter Packet Gap
+#define ECTRL_SLU        0x40   //set link up
```
Comment (shm0sby) [inline kernel/network/e1000.h:76]: Missing space between comment and word

File: kernel/network/ipv4.h [hunk 2]
```diff
@@ -25,7 +25,7 @@
 // #define IPV4_DEBUG
 
 #define IPV4_ADDR_LEN 4
-#define OWN_IPV4_ADDR 0x0A00020A // 10.0.2.10
-#define WAIT_ON_ARP_TIMEOUT 5 //seconds
+#define OWN_IPV4_ADDR 0x0A00020A        // 10.0.2.10
+#define WAIT_ON_ARP_TIMEOUT 5   //seconds
```
Comment (shm0sby) [inline kernel/network/ipv4.h:29]: Shouldnt they be on the same indentation level?

File: kernel/network/pdu_handler.c [hunk 3]
```diff
@@ -27,20 +27,20 @@
 #include <stdio.h>
 
 void
-start_pdu_encapsulation(raw_packet_t *buf) 
+start_pdu_encapsulation(raw_packet_t * buf)
 {
-  ethernet_frame_t* frame = (ethernet_frame_t*) &(buf->data);
-  
+  ethernet_frame_t *frame = (ethernet_frame_t *) & (buf->data);
```
Comment (shm0sby) [inline kernel/network/pdu_handler.c]: Is this `&` really supposed to be so far away from the variable? :D

File: kernel/network/e1000.h [hunk 4]
```diff
@@ -107,40 +107,41 @@
...
-#define TCTL_SWXOFF                     (1 << 22)   // Software XOFF Transmission
-#define TCTL_RTLC                       (1 << 24)   // Re-transmit on Late Collision
-
-#define TSTA_DD                         (1 << 0)    // Descriptor Done
-#define TSTA_EC                         (1 << 1)    // Excess Collisions
-#define TSTA_LC                         (1 << 2)    // Late Collision
-#define LSTA_TU                         (1 << 3)    // Transmit Underrun
+#define TCTL_EN                         (1 << 1)        // Transmit Enable
+#define TCTL_PSP                        (1 << 3)        // Pad Short Packets
+#define TCTL_CT_SHIFT                           4       // Collision Threshold
```
Comment (felixauringer) [inline kernel/network/e1000.h]: why are the normal numbers further right than the shifted numbers?

**Review by shm0sby — COMMENTED** [review]
Is this line length fixed? I think it is a bit too short. 🤔 
Further: Can we document the coding style in an .md file? 
I think some indentations seem a bit random (comment indentation e.g.).

File: kernel/network/arp.c [hunk 5]
```diff
@@ -31,82 +31,86 @@
...
-      default:
-        break;
+    case ARP_REQUEST:
+      arp_handle_request(arp_frame);
+      break;
+    case ARP_REPLY:
+      arp_handle_reply(arp_frame);
+      break;
+    default:
+      break;
```
Comment (felixauringer) [inline kernel/network/arp.c]: is this really the gnu style?

File: kernel/network/arp.c [hunk 6]
```diff
@@ -31,82 +31,86 @@
 #include <string.h>
 
 void
-arp_receive(ethernet_frame_t *frame) 
+arp_receive(ethernet_frame_t * frame)
```
Comment (felixauringer) [inline kernel/network/arp.c:80]: I do not like the new pointer style

File: kernel/network/ipv4.h (hunk 2 above)
Comment (felixauringer) [inline kernel/network/ipv4.h:29]: it seems random how much the comments are indented

File: kernel/network/network_task.c [hunk 7]
```diff
@@ -62,19 +62,19 @@ insert_packet (uint8_t *data, size_t len)
...
-raw_packet_t*
-remove_packet (void)
+raw_packet_t *
+remove_packet(void)
 {
   if (recv_queue_start == recv_queue_end)
     return 0;
 
-  raw_packet_t* packet = &recv_packet_queue[recv_queue_start];
+  raw_packet_t *packet = &recv_packet_queue[recv_queue_start];
```
Comment (felixauringer) [inline kernel/network/network_task.c]: why is here no space after `*`?

File: kernel/network/pdu_handler.c (hunk 3 above)
Comment (felixauringer) [inline kernel/network/pdu_handler.c]: This line sums up all my confusion with spaces and `*` / `&`

File: kernel/network/routing.c [hunk 8]
```diff
@@ -80,37 +83,40 @@ arp_table_get_mac(uint32_t ip)
...
         }
-      else if(arp_table[i].entry_time < arp_table[slot].entry_time)
+      else if (arp_table[i].entry_time < arp_table[slot].entry_time)
         slot = i;
     }
-
-  arp_table[slot] = (arp_table_entry_t) {ip, mac, clock_seconds()};
+  /* *INDENT-OFF* */
+  arp_table[slot] = (arp_table_entry_t) { ip, mac, clock_seconds() };
+  /* *INDENT-ON* */
```
Comment (felixauringer) [inline kernel/network/routing.c:131]: What would the proposed indentation of `indent` be?

Comment (sirkrypt0) [reply kernel/network/routing.c:131]:
```c
arp_table[slot] = (arp_table_entry_t)
{
ip, mac, clock_seconds()};
```

Comment (felixauringer) [reply kernel/network/routing.c:131]: Uff

**Review by felixauringer — APPROVED** [review]

**Pull request merged**